content/docs/command-reference/commit.md

```diff
@@ -17,65 +16,56 @@ positional arguments:
-  of the files/directories used by the stage as dependencies or created as
-  outputs are missing from `dvc.yaml`. It is possible to
-  [add missing data to an existing stage](/docs/user-guide/how-to/add-deps-or-outs-to-a-stage),
-  and then `dvc commit` can be used to save outputs to the cache (and update
-  `dvc.lock`)
-
-- It's always possible to manually execute the command or source code used in a
-  stage without DVC (outputs must be unprotected or removed first in certain
-  cases, see `dvc unprotect`). Once the desired result is reached, use
-  `dvc commit` to update the `dvc.lock` file(s) and store changed data to the
-  cache.
-
-Let's take a look at what is happening in the first scenario closely. Normally
-DVC commands like `dvc add`, `dvc repro` or `dvc run` commit the data to the
-<abbr>cache</abbr> after creating or updating a `dvc.lock` or `.dvc` file. What
-_commit_ means is that DVC:
-
-- Computes a hash for the file/directory.
-- Enters the hash value and file name in the `dvc.lock` or `.dvc` file.
-- Tells Git to ignore the file/directory (adding them to `.gitignore`). (Note
-  that if the <abbr>project</abbr> was initialized with no Git support
-  (`dvc init --no-scm`), this does not happen.)
-- Adds the file(s) in question to the cache.
-
-There are many cases where the last step is not desirable (for example rapid
-iterations on an experiment). The `--no-commit` option prevents it (on the
-commands where it's available). The file hash is still computed and added to the
-`dvc.lock` or `.dvc` file, but the actual data is not cached. And this is where
-the `dvc commit` command comes into play: It performs that last step when
+Stores the current contents of files and directories tracked by DVC in the
+<abbr>cache</abbr>, and updates `dvc.lock` or `.dvc` files as needed. This
+forces DVC to accept the contents of tracked data currently in the
+<abbr>workspace</abbr>, even if they have changed. We explore the scenarios in
+which this can be useful next.
+
+DVC commands that track data (`dvc add`, `dvc repro`, `dvc run`) do the
+following for each file or directory in question:
+
+- Save the hash value of the file(s) in the `dvc.lock` or `.dvc` file.
+- Store the file contents in the cache.
+
+The last step can be skipped with the `--no-commit` option of those commands,
+for example when testing/experimenting with data or
+[stages](/doc/command-reference/run). This avoids caching unfinished data. And
+that's where `dvc commit` comes into play: It performs that last step when
 needed.
 
-Note that it's best to avoid the last three scenarios. They essentially
-force-update the `dvc.lock` or `.dvc` files and save data to cache. They are
-still useful, but keep in mind that DVC can't guarantee reproducibility in those
-cases.
+💡 For convenience, a pre-commit Git hook is available to remind you to
+`dvc commit` when needed. See `dvc install` for more info.
+
+Other scenarios include:
```

Comment:
Not sure `Other` applies here? it's more like a detailed explanation?

Reply:
Hmmm they're meant as different scenarios. The part before this is about `add/run/repro --no-commit` + `dvc commit`. The bullets in this list are
- `add *`
- force-accepting cosmetic changes to dependencies
- adding missing deps/outs
- executing commands manually (this one I guess is pretty similar to the main case)

Reply:
some of them (all of them?) are part of the _main_ use case in this terminology.

`add/run/repro --no-commit` is not even a use case by itself, right? it doesn't explain a specific need.

Reply:
I thought the main reason `commit `exists as a stand-alone command is to complement the `--no-commit` options of add/run/repro?

In terms of the "story", its explained as "forces DVC to accept the contents of tracked data currently in the workspace" in the first p of the description. So do you mean they're all different flavors of that explanation and that there should be a single bullet list (including `add/run/repro --no-commit`)?

Reply:
> I thought the main reason commit exists as a stand-alone command is to complement the --no-commit options of add/run/repro?

probably initially it was the case, now we reference it in other places? (like `--no-exec`?) So we might need to revisit,  generalize? (not 100% sure, just asking)

> So do you mean they're all different flavors of that explanation 

it seems so (not sure about _all_)

> that there should be a single bullet list 

ah, not necessarily. Just removing Other might help? or rephrasing it a bit if the first paragraph is already general.

again, not very constructive feedback here - just highlighting stuff as I read it, a think that seemed strange